tests/Unit/ParallelAlgorithms/LinearSolver/Schwarz/Test_SchwarzAlgorithm.cpp

```diff
@@ -51,143 +134,45 @@ struct SubdomainElementOperator {
+          1, tmpl::list<Tag>>& subdomain_data,
       const gsl::not_null<ResultData*> result,
       const gsl::not_null<SubdomainOperator*> /*subdomain_operator*/) noexcept {
     const size_t element_index = helpers_distributed::get_index(element.id());
+    const size_t num_elements = matrix_slices.size();
+    const size_t num_points_per_element = matrix_slices.begin()->columns();
 
-    // Get the operator matrix slice for the central element
-    const auto& operator_slice = gsl::at(linear_operator, element_index);
-    const size_t num_points = operator_slice.columns();
-    const Index<1> volume_extents{num_points};
-
-    // Apply contribution to the central element
-    const Matrix operator_matrix_element = blaze::submatrix(
-        operator_slice, element_index * num_points, 0, num_points, num_points);
-    apply_matrices(make_not_null(&result->element_data),
-                   make_array<1>(operator_matrix_element),
-                   subdomain_data.element_data, volume_extents);
-
-    // Apply contribution to the overlaps
-    for (const auto& [overlap_id, overlap_data] : subdomain_data.overlap_data) {
-      // Silence unused-variable warning on GCC 7
-      (void)overlap_data;
-      const auto& direction = overlap_id.first;
-      const size_t overlapped_element_index = direction.side() == Side::Lower
-                                                  ? (element_index - 1)
-                                                  : (element_index + 1);
-      const Matrix operator_matrix_overlap = blaze::submatrix(
-          operator_slice, overlapped_element_index * num_points, 0, num_points,
-          num_points);
-      const auto overlap_contribution_extended =
-          apply_matrices(make_array<1>(operator_matrix_overlap),
-                         subdomain_data.element_data, volume_extents);
-      const auto& overlap_extents =
-          gsl::at(all_intruding_extents, direction.dimension());
-      const auto direction_from_neighbor = direction.opposite();
-      LinearSolver::Schwarz::data_on_overlap(
-          make_not_null(&result->overlap_data[overlap_id]),
-          overlap_contribution_extended, volume_extents, overlap_extents,
-          direction_from_neighbor);
-    }
-  }
-};
-
-template <typename Directions>
-struct SubdomainFaceOperator;
+    // Assemble full operator matrix
+    const auto operator_matrix = combine_matrix_slices(matrix_slices);
 
-template <>
-struct SubdomainFaceOperator<domain::Tags::InternalDirections<1>> {
-  using argument_tags = tmpl::list<
-      helpers_distributed::LinearOperator, domain::Tags::Element<1>,
-      domain::Tags::Direction<1>,
-      LinearSolver::Schwarz::Tags::IntrudingExtents<1, SchwarzSmoother>>;
-  using volume_tags = tmpl::list<
-      helpers_distributed::LinearOperator, domain::Tags::Element<1>,
-      LinearSolver::Schwarz::Tags::IntrudingExtents<1, SchwarzSmoother>>;
+    // Extend subdomain data with zeros outside the subdomain
```

Comment:
The combination of the comments and the factoring make this test far easier to read :+1: